src/page-multi-facility/ScenarioLibraryModal.tsx

```diff
@@ -154,32 +163,47 @@ type Props = Pick<ModalProps, "trigger">;
 const ScenarioLibraryModal: React.FC<Props> = ({ trigger }) => {
   const [modalOpen, setModalOpen] = useState(false);
   const [currentScenario, dispatchScenarioUpdate] = useScenario();
-  const [scenarios, setScenarios] = useState({
-    data: [] as Scenario[],
-    loading: true,
-  });
+  const [allScenarios, setAllScenarios] = useState([] as any[]);
   const [baselineScenario, setBaselineScenario] = useState<Scenario | null>();
   const [scenarioIdPendingDeletion, setScenarioIdPendingDeletion] = useState<
     string | null
   >();
   const [showDeleteModal, setShowDeleteModal] = useState(false);
   const rejectionToast = useRejectionToast();
+  const currentUserId = useCurrentUserId();
 
-  async function fetchScenarios() {
+  const fetchScenarios = useCallback(async () => {
     const scenariosData = await getScenarios();
     if (scenariosData) {
-      setScenarios({
-        data: scenariosData,
-        loading: false,
+      const oScenarios = {
+        data: [] as Scenario[],
+        loading: true,
+      };
+      const sScenarios = {
+        data: [] as Scenario[],
+        loading: true,
+      };
+
+      scenariosData.forEach((scenario) => {
+        if (currentUserId && scenario.roles[currentUserId] === "owner") {
+          oScenarios.data.push(scenario);
+        } else {
+          sScenarios.data.push(scenario);
+        }
       });
 
+      oScenarios.loading = false;
+      sScenarios.loading = false;
+
+      setAllScenarios([oScenarios, sScenarios]);
+
       setBaselineScenario(scenariosData.find((scenario) => scenario.baseline));
```

Comment:
I think there is a subtle bug in this line.  If the user deletes the scenario they are currently viewing, they should be taken back to _their_ baseline scenario.  At the moment, when a user deletes the scenario they are currently viewing, they are taken back to the first baseline scenario found in the `scenarioData` which may or may not be the user's baseline scenario.

Reply:
I _think_ the fix for this is roughly:

```javascript
setBaselineScenario(scenariosData.find((scenario) => scenario.baseline && scenario.roles[currentUserId] === "owner"));
```

Reply:
updated!